fe/fe-core/src/main/java/com/starrocks/analysis/DecimalLiteral.java

```diff
@@ -400,11 +400,33 @@ public static void checkLiteralOverflow(BigDecimal value, ScalarType scalarType)
         }
     }
 
+    // for Predicates that contain constant operators of decimal type, the constant value is transferred to
+    // to BE in string type, and in BE, an corresponding VectorizedLiteral is constructed after string value
+    // is converted to decimal via the string-to-decimal casting function who checks decimal overflowing in
+    // decimal style. but in FE, checking decimal overflow in binary style instead of decimal style would
+    // given an incorrect result that overflow checking should fail(in decimal style) expectedly but succeeds
+    // (in decimal style)actually. When checkLiteralOverflowInDecimalStyle fails, proper cast exprs are interpolated
+    // into Predicates to cast the type of decimal constant value to a type wider enough to holds the value.
+    public static void checkLiteralOverflowInDecimalStyle(BigDecimal value, ScalarType scalarType) throws AnalysisException {
```

Comment:
I still don't understand why we need check by binary overflow and check by decimal overflow logic at the same time.

Reply:
When CastExpr and ArithmeticExpr invokes decimal types, checking overflow in binary type is used. When Predicate invokes decimal types, checking overflow in decimal type is used. examples are showed in Description.

Reply:
When CastExpr and ArithmeticExpr invokes decimal types, checking overflow in binary type is used. When Predicate invokes decimal types, checking overflow in decimal type is used. examples are showed in Description.